firebase_admin/_messaging_utils.py

```diff
@@ -613,19 +746,104 @@ def encode_android_notification(cls, notification):
+
+        event_time = result.get('event_time')
+        if event_time:
+            result['event_time'] = str(event_time.isoformat()) + 'Z'
+
+        priority = result.get('notification_priority')
+        if priority and priority not in ('min', 'low', 'default', 'high', 'max'):
+            raise ValueError('AndroidNotification.priority must be "default", "min", "low", "high" '
+                             'or "max".')
+        if priority:
+            result['notification_priority'] = 'PRIORITY_' + priority.upper()
+
+        visibility = result.get('visibility')
+        if visibility and visibility not in ('private', 'public', 'secret'):
+            raise ValueError(
+                'AndroidNotification.visibility must be "private", "public" or "secret".')
+        if visibility:
+            result['visibility'] = visibility.upper()
+
+        vibrate_timings_millis = result.get('vibrate_timings')
+        if vibrate_timings_millis:
+            vibrate_timings_secs = []
+            for msec in vibrate_timings_millis:
+                formated_string = cls.encode_milliseconds(
+                    'AndroidNotification.vibrate_timings_millis', msec)
+                vibrate_timings_secs.append(formated_string)
+            result['vibrate_timings'] = vibrate_timings_secs
+        return result
+
+    @classmethod
+    def encode_light_settings(cls, light_settings):
+        """Encodes a LightSettings instance into JSON."""
+        if light_settings is None:
+            return None
+        if not isinstance(light_settings, LightSettings):
+            raise ValueError(
+                'AndroidNotification.light_settings must be an instance of LightSettings class.')
+        result = {
+            'color': _Validators.check_string(
+                'LightSettings.color', light_settings.color, non_empty=True),
+            'light_on_duration': cls.encode_milliseconds(
+                'LightSettings.light_on_duration_millis', light_settings.light_on_duration_millis),
+            'light_off_duration': cls.encode_milliseconds(
+                'LightSettings.light_off_duration_millis',
+                light_settings.light_off_duration_millis),
+        }
+        result = cls.remove_null_values(result)
+        color = result.get('color')
+        if not color:
+            raise ValueError('LightSettings.color is required.')
+        light_on_duration = result.get('light_on_duration')
+        if not light_on_duration:
+            raise ValueError(
+                'LightSettings.light_on_duration_millis is required.')
+        light_off_duration = result.get('light_off_duration')
+        if not light_off_duration:
+            raise ValueError(
+                'LightSettings.light_off_duration_millis is required.')
+
+        if not re.match(r'^#[0-9a-fA-F]{6}$', color) and not re.match(r'^#[0-9a-fA-F]{8}$', color):
+            raise ValueError(
+                'LightSettings.color must be in the form #aabbcc or aabbccdd.')
```

Comment:
Was there a reason to not use "#rrggbb" or "#rrggbbaa"?

Reply:
Fixed. Thanks!